From 71cdf69642de9036939bd58c5fa6a1701a707ff5 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Wed, 19 Oct 2016 16:47:14 +0300
Subject: [PATCH 0539/1345] doc: a7040: a8040: update documentation for
 A8040-DB & A7040-DB boards

commit  b6a8abbe9c47019120c08c7db6f5491d9ca3058a from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- add new file armada8040-db-setup with single board configuration.
- update and align armada7040-db-setup.

Change-Id: Ie9c07fdd36247fac08e715fcf185ac6c8cb92af6
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33306
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../mvebu/a7k-a8k/armada7040-db-setup.txt          |   49 +++++++----
 .../mvebu/a7k-a8k/armada8040-db-setup.txt          |   92 ++++++++++++++++++++
 2 files changed, 123 insertions(+), 18 deletions(-)
 create mode 100644 Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt

diff --git a/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
index 056445a..ecac1d9 100644
--- a/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
+++ b/Documentation/mvebu/a7k-a8k/armada7040-db-setup.txt
@@ -1,13 +1,21 @@
 Armada7040 development board setup:
 --------------------------------------
 
-The Armada7040 Development Board (DB) is a configurable board designed to test several
-combinations of SoC configurations. Linux supports a defined set of those configurations
-via different device tree files.
+the Armada 7040 development board (PCB: DB-88F7040-Modular) is a configurable board,
+designed to test several combinations of SoC configurations.
+linux supports a defined set of those configurations via different device tree files.
 
 This document describes the board modifications required to set up each configuration and the
 interfaces supported (or disabled in each one).
 
+To switch board configuration:
+1. Modify HW board settings according to the instructions in the "Board Setup" section below.
+2. Use appropriate Device Tree file for booting Linux.
+3. Same Device Tree selection is required also in U-Boot (to initialize requested comphy SerDes lanes).
+   Modify the Device Tree file used by U-BOOT, by running the U-BOOT command "fdt_config",
+   and select the appropriate configuration entry.
+
+
 Board Setup
 ------------
 
@@ -49,7 +57,6 @@ The tables below summarizes the interface configuration of each setup
 
 SERDES PHY configuration
 ------------------------
-
 Setup  #	|	1	|	4	| 	6	|
 Name   #	|     Router 	|     NAND	|     RXAUI	|
 -----------------------------------------------------------------
@@ -64,21 +71,27 @@ SERDES:	lane0	| sgmii2	| pex_x4_l0	|  sata1	|
 
 Multi-purpose pin configurations
 --------------------------------
-
 Setup  #	|	1	|       4	| 	6	|
 Name   #	|     Router 	|     NAND	|     RXAUI	|
 -----------------------------------------------------------------
-
-MPPs: 	CP-SPI0	| N/C		| N/C		|  N/C		|
-	CP-SPI1	| [13-16]	| N/C		|  [13-16]	|
-	NAND	| N/C		| [13,17-27]	|  N/C		|
-	RGMII0	| N/C		| [0-12]	|  N/C		|
-	RGMII1	| N/C		| [44-55]	|  N/C		|
-	CP-UART0| [29-30]	| [29-30]	|  [29-30]	|
-	CP-UART1| [40-41]	| [40-41]	|  [40-41]	|
-	SD	| [56-62]	| [56-62]	|  [56-62]	|
-	TDM	| [0-11]	| N/C		|  N/C		|
-	TWSI	| [37-38]	| [37-38]	|  [37-38]	|
-	SATA0	| [36]		| N/C		|  [36]		|
-	SATA1	| [28]		| [28]		|  [28]		|
+AP806:								|
+-----------------------------------------------------------------
+	AP-SDIO | [0-10]	| [0-10]	| [0-10]	|
+	AP-SPI0 | N/C		| N/C		| N/C		|
+	AP-UART0| [11,19]	| 11,19]	| [11,19]	|
+-----------------------------------------------------------------
+CP:								|
+-----------------------------------------------------------------
+	CP-SPI0	| N/C		| N/C		| N/C		|
+	CP-SPI1	| [13-16]	| N/C		| [13-16]	|
+	NAND	| N/C		| [13,17-27]	| N/C		|
+	RGMII0	| N/C		| [0-12]	| N/C		|
+	RGMII1	| N/C		| [44-55]	| N/C		|
+	CP-UART0| [29-30]	| [29-30]	| [29-30]	|
+	CP-UART1| [40-41]	| [40-41]	| [40-41]	|
+	SD	| [56-62]	| [56-62]	| [56-62]	|
+	TDM	| [0-11]	| N/C		| N/C		|
+	TWSI	| [37-38]	| [37-38]	| [37-38]	|
+	SATA0	| [36]		| N/C		| [36]		|
+	SATA1	| [28]		| [28]		| [28]		|
 -----------------------------------------------------------------
diff --git a/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
new file mode 100644
index 0000000..275350aa
--- /dev/null
+++ b/Documentation/mvebu/a7k-a8k/armada8040-db-setup.txt
@@ -0,0 +1,92 @@
+Armada8040 development board setup:
+--------------------------------------
+
+The Armada 8040 development board (PCB: DB-88F8040-Modular) is a configurable board,
+designed to test several combinations of SoC configurations.
+linux supports a defined set of those configurations via different device tree files.
+
+This document describes the board modifications required to set up each configuration and the
+interfaces supported (or disabled in each one).
+
+To switch board configuration:
+1. Modify HW board settings according to the instructions in the "Board Setup" section below.
+2. Use appropriate Device Tree file for booting Linux.
+3. Same Device Tree selection is required also in U-Boot (to initialize requested comphy SerDes lanes).
+   Modify the Device Tree file used by U-BOOT, by running the U-BOOT command "fdt_config",
+   and select the appropriate configuration entry.
+
+
+Board Setup
+------------
+
+## setup 1: Default ###
+
+   - Serdes CP0 site module (CON42): SLM1521
+   - Serdes CP1 site module (CON41): SLM1521
+   - Jumper-module (CON44): SLM1564
+   - SLM1521 Module DIP switch:
+	- DPR 2,3,4,5 = 1-2
+	- SW1 = ON
+	- SW2 = OFF
+	- SW3 = ON
+	- SW4 = OFF
+	- SW5 = OFF
+	- SW6 - ON
+	- SW7 - ON
+	- SW8 - OFF
+
+
+The tables below summarizes the interface configuration of each setup
+
+SERDES PHY configuration
+------------------------
+Setup  #	|	1	|	TBD	| 	TBD	|
+Name   #	|     Default 	|		|		|
+-----------------------------------------------------------------
+CP0:								|
+-----------------------------------------------------------------
+	lane0	| sgmii2	|		|		|
+	lane1	| usb3h0	|		|		|
+	lane2	| KR (10G)	|		|		|
+	lane3	| sata1		|		|		|
+	lane4	| usb3h1	|		|		|
+	lane5	| pex_x1_2	|		|		|
+-----------------------------------------------------------------
+CP1:								|
+-----------------------------------------------------------------
+	lane0	| sgmii2	|		|		|
+	lane1	| usb3h0	|		|		|
+	lane2	| KR (10G)	|		|		|
+	lane3	| sata1		|		|		|
+	lane4	| N/C		|		|		|
+	lane5	| pex_x1_2	|		|		|
+-----------------------------------------------------------------
+
+
+
+Multi-purpose pin configurations
+--------------------------------
+Setup  #	|	1	|	TBD	|	TBD	|
+Name   #	|     Default 	|		|		|
+-----------------------------------------------------------------
+AP806:								|
+-----------------------------------------------------------------
+    AP-SDIO	| N/C		|		|		|
+    AP-SPI0	| [4-5]		|		|		|
+    AP-UART0	| [11,19]	|		|		|
+-----------------------------------------------------------------
+CP0:								|
+-----------------------------------------------------------------
+    CP0-SMI	| [32,34]	|		|		|
+    CP0-XSMI	| [35-36]	|		|		|
+    CP0-TWSI	| [37-38]	|		|		|
+    CP0-SATA1	| [40]		|		|		|
+    CP0-SATA0	| [41]		|		|		|
+    CP0-RGMII1	| [44-55]	|		|		|
+    CP0-SD	| [56-62]	|		|		|
+-----------------------------------------------------------------
+CP1:								|
+-----------------------------------------------------------------
+    CP1-RGMII1	| [0-11]	|		|		|
+    CP1-SMI	| [27,31]	|		|		|
+-----------------------------------------------------------------
-- 
1.7.9.5

